src/syrupy/__init__.py

```diff
@@ -114,15 +114,17 @@ def pytest_collection_modifyitems(session: Any, config: Any, items: List[Any]) -
     After tests are collected and before any modification is performed.
     https://docs.pytest.org/en/latest/reference.html#_pytest.hookspec.pytest_collection_modifyitems
     """
-    config._syrupy._all_items.update(items)
+    valid_items = (item for item in session.items if hasattr(item, "obj"))
```

Comment:
Shouldn't this be `items`
```suggestion
    valid_items = (item for item in items if hasattr(item, "obj"))
```